src/Umbraco.Web.UI.Client/src/common/services/iconhelper.service.js

```diff
@@ -88,12 +88,59 @@ function iconHelper($http, $q, $sce, $timeout, umbRequestHelper) {
     var collectedIcons;
 
     var imageConverter = [
-            {oldImage: "contour.png", newIcon: "icon-umb-contour"}
-            ];
+        {oldImage: "contour.png", newIcon: "icon-umb-contour"}
+    ];
 
     var iconCache = [];
-    var liveRequests = [];
-    var allIconsRequested = false;
+    var resourceLoadStatus = "none";
+    var promiseQueue = [];
+
+    /**
+     * This is the same approach as use for loading the localized text json 
+     * We don't want multiple requests for the icon collection, so need to track
+     * the current request state, and resolve the queued requests once the icons arrive
+     * Subsequent requests are returned immediately as the icons are cached into 
+     */
+    function init() {       
+        var deferred = $q.defer();
+
+        if (resourceLoadStatus === "loaded") {
+            deferred.resolve(iconCache);
+            return deferred.promise;
+        }
+
+        if (resourceLoadStatus === "loading") {
+            promiseQueue.push(deferred);
+            return deferred.promise;
+        }
+
+        resourceLoadStatus = "loading";
+
+        $http({ method: "GET", url: Umbraco.Sys.ServerVariables.umbracoUrls.iconApiBaseUrl + 'GetIcons' })
+            .then(function (response) {
+                resourceLoadStatus = "loaded";
+                var resourceLoadingPromise = [];
```

Comment:
minor: but `const` is preferred over `var`

Reply:
`resourceLoadingPromise` is never populated so not sure what this is doing?

Reply:
@Shazwazza `resourceLoadingPromise` should have been renamed to `promiseQueue`, to use a more descriptive name. The queue is populated when simultaneous icon requests are made, then all are resolved when the icons arrive. 

I've updated to resolve your other comments, should be all good now.